ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -319,6 +322,20 @@ object BigDiffy {
     (r: TableRow) => get(xs, 0, r)
   }
 
+  def pathWithShards(path: String): String = path.replaceAll("\\/+$", "") + "/part"
+
+  implicit class TextFileHeader(coll: SCollection[String]) {
+    def saveAsTextFileWithHeader(path: String, header: String): Future[Tap[String]] = {
+      val transform = TextIO.write()
+        .to(pathWithShards(path))
+        .withSuffix(".txt")
```

Comment:
do we want this to be `.tsv` since that's what it produces

Reply:
Theoretically it could be, but the existing code outputs `.txt` and I'm moving this header to be optional. I'd rather keep the file format/naming consistent.